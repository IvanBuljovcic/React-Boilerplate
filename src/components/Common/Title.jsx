import React from 'react';
import PropType from 'prop-types';

const Title = ({ text, isSubtitle }) => {
  if (isSubtitle) {
    return (
      <h2 display-if={text} className="app-title app-title--subtitle">
        {text}
      </h2>
    );
  }

  return (
    <h1 display-if={text} className="app-title">
      {text}
    </h1>
  );
};

Title.propTypes = {
  text: PropType.string,
  isSubtitle: PropType.bool
};

Title.defaultProps = {
  text: null,
  isSubtitle: false
};

export default Title;
