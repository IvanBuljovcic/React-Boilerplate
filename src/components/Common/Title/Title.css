.app-title {
  color: #ccc;
  font-size: 2.4rem;

  &--modifier {
    color: #fa0;
  }
}
