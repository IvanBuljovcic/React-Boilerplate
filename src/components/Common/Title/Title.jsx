import React from 'react';
import PropTypes from 'prop-types';

import './Title.css';

const Title = ({ text, isSubtitle }) => {
  if (isSubtitle) {
    return (
      <h2 display-if={text} className="app-title app-title--subtitle">
        {text}
      </h2>
    );
  }

  return (
    <h1 display-if={text} className="app-title">
      {text}
    </h1>
  );
};

Title.propTypes = {
  text: PropTypes.string,
  isSubtitle: PropTypes.bool
};

Title.defaultProps = {
  text: null,
  isSubtitle: false
};

export default Title;
