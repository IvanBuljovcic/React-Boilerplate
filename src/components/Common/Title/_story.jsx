import React from 'react';
import { storiesOf } from '@storybook/react';

// - Component
import Title from './Title';

const stories = storiesOf('Common', module);

stories.addWithChapters('Title', {
  chapters: [
    {
      sections: [
        {
          title: 'Default',
          subtitle: `Default title, accepts a **text** prop`,
          sectionFn: () => <Title text="Hello World" />
        },
        {
          title: 'Subtitle',
          info: 'Subtitle title, accepts a **text** prop',
          sectionFn: () => <Title text="Hello World Subtitle" isSubtitle />
        },
        {
          title: 'With Modifier',
          info: 'Title with a class modifier. The modifier appends a class to the element with the **app-title--modifier** pattern',
          sectionFn: () => <Title text="Hello World modified" classModifier="modifier" />
        }
      ]
    }
  ]
});
