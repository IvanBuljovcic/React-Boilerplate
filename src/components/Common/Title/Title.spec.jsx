import React from 'react';
import renderer from 'react-test-renderer';

// - Test component
import Title from './Title';

describe('<Title /> component renders correctly', () => {
  test('Default', () => {
    const tree = renderer.create(<Title text="My title" />).toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('Subtitle', () => {
    const tree = renderer.create(<Title text="My subtitle" isSubtitle />).toJSON();

    expect(tree).toMatchSnapshot();
  })
});

