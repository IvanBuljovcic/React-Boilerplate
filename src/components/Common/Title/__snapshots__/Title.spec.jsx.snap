// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> component renders correctly Default 1`] = `
<h1
  className="app-title"
>
  My title
</h1>
`;

exports[`<Title /> component renders correctly Subtitle 1`] = `
<h2
  className="app-title app-title--subtitle"
>
  My subtitle
</h2>
`;
